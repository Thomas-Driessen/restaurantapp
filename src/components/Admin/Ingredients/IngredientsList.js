import React from 'react';
import Grid from '@material-ui/core/Grid';
import Ingredient from './Ingredient'

const IngredientsList = (props) => {
    return (
        <Grid container spacing={1} style={{ padding: 10 }}>
            { props.ingredients.map((currentIngredient, index) => (
                <Grid key={index} item xs={12} sm={6} lg={4} xl={3}>
                    <Grid item>
                        <Ingredient
                            ingredient={currentIngredient}
                        />
                    </Grid>
                </Grid>
            ))}
        </Grid>
    )
}
export default IngredientsList