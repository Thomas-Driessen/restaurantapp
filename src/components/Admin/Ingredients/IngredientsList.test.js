import React from 'react';
import { render } from '@testing-library/react';
import IngredientsList from './IngredientsList';
import renderer from 'react-test-renderer';

const ingredients = [
  {
    ingredientId: 1,
    ingredientTitle: "Salt",
    ingredientQuantity: 400,
    price: 2.5,
  },
  {
    ingredientId: 2,
    ingredientTitle: "Vinegar",
    ingredientQuantity: 400,
    price: 2.5,
  },
];

test('Render Ingredient Salt', () => {
  const { getByText } = render(<IngredientsList ingredients={ingredients} />);
  const linkElement = getByText(/Salt/i);
  expect(linkElement).toBeInTheDocument();
});

test('Render Ingredient Vinegar', () => {
  const { getByText } = render(<IngredientsList ingredients={ingredients} />);
  const linkElement = getByText(/Vinegar/i);
  expect(linkElement).toBeInTheDocument();
});

it('IngredientsList renders correctly', () => {
  const tree = renderer
    .create(<IngredientsList ingredients={ingredients} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

afterAll(() => {
  global.gc && global.gc()
})