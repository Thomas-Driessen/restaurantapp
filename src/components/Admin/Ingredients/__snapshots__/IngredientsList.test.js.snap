// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IngredientsList renders correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
  style={
    Object {
      "padding": 10,
    }
  }
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-4 MuiGrid-grid-xl-3"
  >
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <div>
        <div>
          <div
            className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              className="MuiCardContent-root"
            >
              <h2
                className="MuiTypography-root MuiTypography-gutterBottom"
              >
                Salt
                 
                <span
                  style={
                    Object {
                      "float": "right",
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "color": "green",
                      }
                    }
                  >
                     
                    2.5
                    € 
                  </span>
                   / 100g
                </span>
              </h2>
              <h6
                className="MuiTypography-root MuiTypography-body1"
              >
                Quantity: 
                400
              </h6>
            </div>
            <div
              className="MuiCardActions-root MuiCardActions-spacing"
            >
              <div
                className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-flex-start MuiGrid-justify-xs-flex-end"
              >
                <div
                  style={
                    Object {
                      "alignItems": "right",
                      "display": "flex",
                    }
                  }
                >
                  <div>
                    <button
                      aria-label="edit"
                      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <span
                        className="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                  <div>
                    <button
                      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <span
                        className="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 13H5v-2h14v2z"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-4 MuiGrid-grid-xl-3"
  >
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <div>
        <div>
          <div
            className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              className="MuiCardContent-root"
            >
              <h2
                className="MuiTypography-root MuiTypography-gutterBottom"
              >
                Vinegar
                 
                <span
                  style={
                    Object {
                      "float": "right",
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "color": "green",
                      }
                    }
                  >
                     
                    2.5
                    € 
                  </span>
                   / 100g
                </span>
              </h2>
              <h6
                className="MuiTypography-root MuiTypography-body1"
              >
                Quantity: 
                400
              </h6>
            </div>
            <div
              className="MuiCardActions-root MuiCardActions-spacing"
            >
              <div
                className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-flex-start MuiGrid-justify-xs-flex-end"
              >
                <div
                  style={
                    Object {
                      "alignItems": "right",
                      "display": "flex",
                    }
                  }
                >
                  <div>
                    <button
                      aria-label="edit"
                      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <span
                        className="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                  <div>
                    <button
                      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <span
                        className="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 13H5v-2h14v2z"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
